src/Core/Silk.NET.Core/Native/Structs/IUnknown.cs

```diff
@@ -0,0 +1,85 @@
+﻿// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using Silk.NET.Core.Native;
+using Silk.NET.Core.Attributes;
+using Silk.NET.Core.Contexts;
+using Silk.NET.Core.Loader;
+
+#pragma warning disable 1591
+
+namespace Silk.NET.Core.Native
+{
+    public unsafe partial struct IUnknown
+    {
+        public void** LpVtbl;
+        /// <summary>To be added.</summary>
+        public readonly int QueryInterface(Guid* riid, void** ppvObject)
+        {
+            var @this = (IUnknown*) Unsafe.AsPointer(ref Unsafe.AsRef(in this));
+            return ((delegate* unmanaged[Cdecl]<IUnknown*, Guid*, void**, int>) LpVtbl[0])(@this, riid, ppvObject);
+        }
+
+        /// <summary>To be added.</summary>
+        public readonly int QueryInterface(Guid* riid, ref void* ppvObject)
+        {
+            var @this = (IUnknown*) Unsafe.AsPointer(ref Unsafe.AsRef(in this));
+            fixed (void** ppvObjectPtr = &ppvObject)
+            {
+                return ((delegate* unmanaged[Cdecl]<IUnknown*, Guid*, void**, int>) LpVtbl[0])
+                    (@this, riid, ppvObjectPtr);
+            }
+        }
+
+        /// <summary>To be added.</summary>
+        public readonly int QueryInterface(ref Guid riid, void** ppvObject)
+        {
+            var @this = (IUnknown*) Unsafe.AsPointer(ref Unsafe.AsRef(in this));
+            fixed (Guid* riidPtr = &riid)
+            {
+                return ((delegate* unmanaged[Cdecl]<IUnknown*, Guid*, void**, int>) LpVtbl[0])
+                    (@this, riidPtr, ppvObject);
+            }
+        }
+
+        /// <summary>To be added.</summary>
+        public readonly int QueryInterface(ref Guid riid, ref void* ppvObject)
+        {
+            var @this = (IUnknown*) Unsafe.AsPointer(ref Unsafe.AsRef(in this));
+            fixed (Guid* riidPtr = &riid)
+            {
+                fixed (void** ppvObjectPtr = &ppvObject)
+                {
+                    return ((delegate* unmanaged[Cdecl]<IUnknown*, Guid*, void**, int>) LpVtbl[0])
+                        (@this, riidPtr, ppvObjectPtr);
+                }
+            }
+        }
+
+        /// <summary>To be added.</summary>
```

Comment:
idk why you've started to do this, but don't. Either add something like "To be documented." or just have the warning. "To be added." confuses users.

Reply:
This comment was chosen to remain consistent with the rest of Silk.NET.

Reply:
fair. will put up a PR to change this everywhere after this has been merged.